src/wallet/wallet.cpp

```diff
@@ -2198,111 +2198,109 @@ CAmount CWallet::GetAvailableBalance(const CCoinControl* coinControl) const
 
 void CWallet::AvailableCoins(std::vector<COutput> &vCoins, bool fOnlySafe, const CCoinControl *coinControl, const CAmount &nMinimumAmount, const CAmount &nMaximumAmount, const CAmount &nMinimumSumAmount, const uint64_t nMaximumCount, const int nMinDepth, const int nMaxDepth) const
 {
+    AssertLockHeld(cs_main);
+    AssertLockHeld(cs_wallet);
```

Comment:
Both are recursive locks, so any reason we can't just use `LOCK2` here?

Reply:
Long term plan is to remove recursive locks. Correct me if I'm wrong @TheBlueMatt.

Reply:
I think in general we want to move away from recursive locking: https://github.com/bitcoin/bitcoin/blob/85123be78df5a4e9447e70f9ee727416d45843d7/src/sync.h#L92

http://www.zaval.org/resources/library/butenhof1.html has some interesting rationale and history

Reply:
Thanks @ryanofsky for the reference.

Reply:
Ah, thanks for the clarification.